server.py

```diff
@@ -35,29 +41,33 @@ def log_reponse_info(response):
 
 # protected/non-protected routes
 open_routes = {
-    ('/login', 'POST'): 'login',
-    ('/register', 'POST'): 'register',
-    ('/create_bank_account', 'POST'): 'create_bank_account'
+    ('/users/login', 'POST'): 'login',
+    ('/users/register', 'POST'): 'register',
+    ('/bank_accounts', 'POST'): 'create_bank_account'
     }
 
 closed_routes = {
-    ('/get_user', 'GET'): 'get_user',
-    ('/bank_accounts', 'GET') : 'get_bank_account'
+    ('/users', 'GET'): 'get_user',
+    ('/bank_accounts', 'GET') : 'get_bank_account',
+    ('/bank_accounts/transfer', 'POST') : 'transfer'
 }
 
 # Middleware for token auth
 @app.before_request
 def auth_token():
-    # Normalize the request path (strip any trailing slashes) and lowercase it for comparison
     request_path = request.path.rstrip('/').lower()
-    request_key = (request_path, request.method)
+    request_method = request.method
+    request_key = (request_path, request_method)
     logging.debug(f"Request key: {request_key}")
+    logging.debug(f"Open routes: {open_routes}")
+    logging.debug(f"Closed routes: {closed_routes}")
+    logging.debug(f"All registered routes: {[str(rule) for rule in app.url_map.iter_rules()]}")
     
-    # check if there's a match between path and method
     if request_key in open_routes:
+        logging.debug(f"Matched open route: {request_key}")
         return
-
     elif request_key in closed_routes:
```

Comment:
What if it's a round that not open and not closed? the whole idea is to NOT need to add all the API routes and let the middleware apply authentication by default, UNLESS you choose to bypass a specific route.